pkgs/coverage/lib/src/coverage_options.dart

```diff
@@ -0,0 +1,124 @@
+import 'dart:io';
+import 'package:cli_config/cli_config.dart';
+
+class CoverageOptions {
+  const CoverageOptions({
+    required this.output,
+    this.connectTimeout,
+    required this.scopeOutput,
+    required this.resumeIsolates,
+    required this.functionCoverage,
+    required this.branchCoverage,
+    required this.packagePath,
+    this.input,
+    this.reportOn,
+    required this.workers,
+    this.baseDirectory,
+    required this.prettyPrint,
+    required this.lcov,
+    required this.ignoreFiles,
+    this.packageName,
+    required this.testScript,
+    required this.verbose,
+  });
+
+  factory CoverageOptions.fromConfig(
+      Config options, CoverageOptions defaultOptions) {
+    return CoverageOptions(
```

Comment:
Like I said [on the bug](https://github.com/dart-lang/tools/issues/532#issuecomment-2567248818), I think we don't need most of these options in the yaml. Remove `connect_timeout`, `resume_isolates`, `in`, `report_on`, `workers`, `base_directory`, `pretty_print`, `lcov`, `ignore_files`, and `verbose`.